packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -95,15 +97,20 @@ const TwoFactorVerifyModal = ({ open, onClose }) => {
     const handleResendCode = async () => {
         setResendCode('resending');
         await Mixpanel.withTracking('2FA Modal Resend code', 
-            async () => await dispatch(resendTwoFactor()),
+            async () => {
+                await dispatch(resendTwoFactor());
+                setResendCode('resent');
+                setTimeout(() => { setResendCode(); }, 3000);
+            },
             (e) => {
-                setResendCode();
+                if (e.status === TOO_MANY_REQUESTS_STATUS) {
+                    onClose(false, e);
```

Comment:
Close the modal when we hit a rate limit, preventing the user for verifying/resending when we know it will fail.

Reply:
`e` here will also get surfaced to the pending action ensuring that the `<GlobalAlert />` is populated with the error message.